superduper/vector_search/qdrant.py

```diff
@@ -0,0 +1,190 @@
+            )
+            for item in items
+        ]
+        self.client.upsert(collection_name=self.collection_name, points=points)
+
+    def delete(self, ids: t.Sequence[str]) -> None:
+        """Delete vectors from the index.
+
+        :param ids: List of IDs to delete
+        """
+        self.client.delete(
+            collection_name=self.collection_name,
+            points_selector=models.Filter(
+                must=[
+                    models.FieldCondition(
+                        key=ID_PAYLOAD_KEY, match=models.MatchAny(any=list(ids))
+                    )
+                ]
+            ),
+        )
+
+    def find_nearest_from_id(
+        self,
+        _id,
+        n: int = 100,
+        within_ids: t.Sequence[str] = (),
+    ) -> t.Tuple[t.List[str], t.List[float]]:
+        """Find the nearest vectors to a given ID.
+
+        :param _id: ID to search
+        :param n: Number of results to return
+        :param within_ids: List of IDs to search within
+        """
+        query_filter = None
+        if within_ids:
+            query_filter = models.Filter(
+                must=[
+                    models.FieldCondition(
+                        key=ID_PAYLOAD_KEY, match=models.MatchAny(any=list(within_ids))
+                    )
+                ]
+            )
+
+        search_result = self.client.query_points(
+            collection_name=self.collection_name,
+            query=self._convert_id(_id),
+            limit=n,
+            query_filter=query_filter,
+            with_payload=[ID_PAYLOAD_KEY],
+            using=self.vector_name,
+        ).points
+
+        ids = [hit.payload[ID_PAYLOAD_KEY] for hit in search_result]  # type: ignore
+        scores = [hit.score for hit in search_result]
+
+        return ids, scores
+
+    def find_nearest_from_array(
+        self,
+        h: np.typing.ArrayLike,
+        n: int = 100,
+        within_ids: t.Sequence[str] = (),
+    ) -> t.Tuple[t.List[str], t.List[float]]:
+        """Find the nearest vectors to a given vector.
+
+        :param h: Vector to search
+        :param n: Number of results to return
+        :param within_ids: List of IDs to search within
+        """
+        query_filter = None
+        if within_ids:
+            query_filter = models.Filter(
+                must=[
+                    models.FieldCondition(
+                        key=ID_PAYLOAD_KEY, match=models.MatchAny(any=list(within_ids))
+                    )
+                ]
+            )
+
+        search_result = self.client.query_points(
+            collection_name=self.collection_name,
+            query=h,
+            limit=n,
+            query_filter=query_filter,
+            with_payload=[ID_PAYLOAD_KEY],
+            using=self.vector_name,
+        ).points
+
+        ids = [hit.payload[ID_PAYLOAD_KEY] for hit in search_result]  # type: ignore
+        scores = [hit.score for hit in search_result]
+
+        return ids, scores
```

Comment:
This section of logic, apart from having a different query, appears to be the same as that in `find_nearest_from_id`. If yes, please create a common function to handle it.

Reply:
[`cbdd7a8` (#2428)](https://github.com/superduper-io/superduper/pull/2428/commits/cbdd7a868db9e538a90b90ab1df16f2e4db41331)